lib/msf/ui/console/command_dispatcher/core.rb

```diff
@@ -1794,6 +1836,84 @@ def cmd_sessions(*args)
     true
   end
 
+  def get_matching_sessions(search_term)
+    matching_sessions = {}
+    terms = search_term.split
+    terms.each do |term|
+      matches = filter_sessions_by_search(term)
+      matches.each do |session_id, session|
+        matching_sessions.store(session_id, session)
+      end
+    end
+    matching_sessions
+  end
+
+  def filter_sessions_by_search(search_term)
+    matching_sessions = {}
+    
+    framework.sessions.each do |session_id, session|
+      case search_term.split(":")[0]
+      when "last_checkin"
+        if session.respond_to?(:last_checkin) && session.last_checkin && evaluate_search_criteria(session, search_term)
+          matching_sessions.store(session_id, session)
+        end
+      when "session_type"
+        matching_sessions.store(session_id, session) if evaluate_search_criteria(session, search_term)
+      when "session_id"
+        matching_sessions.store(session_id, session) if evaluate_search_criteria(session, search_term)
+      end
+    end
+    matching_sessions
+  end
+
+  def evaluate_search_criteria(session, search_term)
+    parts = search_term.split(":")
+
+    field = parts[0]
+    operator = parts[1]
+    value = parts[2] if parts[2]
+    return false unless VALID_PARAMS.include? parts[0]
+
+    case field
+    when "last_checkin"
+      checkin_time = session.last_checkin
+      threshold_time = Time.now - parse_duration(value)
+      case operator
+      when "before"
+        return checkin_time > threshold_time
+      when "after"
+        return checkin_time < threshold_time
+      else
+        return false
+      end
+    when "session_id"
+      return session.sid.to_s == operator
+    when "session_type"
+      return session.type == operator
+    end
+  end
+
+  def parse_duration(duration)
+    total_time = 0
+    time_tokens = duration.scan(/\d+/).zip(duration.scan(/[a-zA-Z]+/))
+    time_tokens.each do |pair|
+      raise "Please specify both time units and amounts" if pair[1].nil?
+      case pair[1]
+      when "d"
+        total_time = total_time + pair[0].to_i * 86400
+      when "h"
+        total_time = total_time + pair[0].to_i * 3600
+      when "m"
+        total_time = total_time + pair[0].to_i * 60
+      when "s"
+        total_time = total_time + pair[0].to_i
+      else
+        raise "Unrecognized time format"
```

Comment:
```suggestion
        raise ArgumentError, "Unrecognized time format: #{value}"
```
Give the user something to work with, maybe even add what values are allowed